### PR TITLE
add streaming job example with google pub/sub connector

### PR DESCRIPTION
**Pull request by geazi-anc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ We suggest to remove the official `flink-scala` and `flink-streaming-scala` depe
 There is a wide range of [code examples](https://github.com/flink-extended/flink-scala-api/tree/master/modules/examples) to introduce you to flink-scala-api, both using Scala scripts and multimodule applications. These examples include:
 
 - Flink jobs built using Scala 3 with Ammonite and Scala CLI;
+- Real time data processing job with Google Pub/Sub connector and custom JSON serializers and deserializers;
```
Comment (novakov-alexey) [inline README.md:58]: Maybe rephrase it to "Streaming job: Google Pub/Sub using JSON serialiser

File: modules/examples/scripts/StreamingJobExampleWithPubSubConnectorAndCustomJsonSerializers.sc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * This code is available in article version with more details of the
+ * implementation:
+ * https://dev.to/geazi_anc/data-engineering-with-scala-mastering-real-time-data-processing-with-apache-flink-and-google-pubsub-3b39
+ */
+
+// model definitions
+final case class CreatedCustomer(fullName: String, birthDate: String) derives ReadWriter:
+  def firstName: String = fullName.split(" ").head
```
Comment (novakov-alexey) [inline modules/examples/scripts/StreamingJobExampleWithPubSubConnectorAndCustomJsonSerializers.sc:24]: all 3 methods can be vals?

**Comment by geazi-anc** [comment]
@novakov-alexey thank you so much for your suggestions! If there are any more, please let me know :)

File: modules/examples/scripts/StreamingJobExampleWithPubSubConnectorAndCustomJsonSerializers.sc [hunk 3]
```diff
@@ -0,0 +1,69 @@
+//> using toolkit default
```
Comment (novakov-alexey) [inline modules/examples/scripts/StreamingJobExampleWithPubSubConnectorAndCustomJsonSerializers.sc:1]: Can you also shorten file name to 3 words or so?
